classes/helper.php

```diff
@@ -652,7 +655,7 @@ private static function check_if_already_acknowledged_by_user(sitenotice $notice
             // The reset interval has been past.
             || (($notice->resetinterval > 0) && ($latestview->timemodified + $notice->resetinterval < time()))
             // The previous action is 'dismiss', so still require acknowledgement.
-            || ($latestview->action === acknowledgement::ACTION_DISMISSED && $notice->reqack == true)) {
+            || ($latestview->action == acknowledgement::ACTION_DISMISSED && $notice->reqack == true)) {
```

Comment:
Why does it need to be changed to `==`? What are the possible values of `$latestview->action` - could we maybe sanitize it and continue to use `===`?

Reply:
See my response to @keevan